rust/src/dns/detect.rs

```diff
@@ -114,9 +115,42 @@ pub unsafe extern "C" fn rs_dns_detect_opcode_free(ptr: *mut c_void) {
     }
 }
 
+/// Perform the DNS rcode match.
+///
+/// 1 will be returned on match, otherwise 0 will be returned.
+#[no_mangle]
+pub extern "C" fn rs_dns_rcode_match(
+    tx: &mut DNSTransaction, detect: &mut DetectUintData<u8>, flags: u8,
+) -> u8 {
+    let header_flags = if flags & Direction::ToServer as u8 != 0 {
+        if let Some(request) = &tx.request {
+            request.header.flags
+        } else {
+            return 0;
+        }
+    } else if flags & Direction::ToClient as u8 != 0 {
+        if let Some(response) = &tx.response {
+            response.header.flags
+        } else {
+            return 0;
+        }
+    } else {
+        // Not to server or to client??
+        return 0;
+    };
+
+    let rcode = (header_flags & 0xf) as u8;
```

Comment:
@jasonish should we use `tx.rcode()` here ?

Reply:
And @hadiqaalamdar how can this be 23 ? cf commit message saying `valid ranges = [0-23]`

Reply:
> @jasonish should we use `tx.rcode()` here ?

That won't get you the rcode for a request, even if its odd to do so.  Could extend it to, or just leave this as is for now.

Reply:
@jasonish is the owner of DNS knowledge and he said

> just leave this as is for now.